src/libs/DateUtils.js

```diff
@@ -215,37 +301,41 @@ function getStatusUntilDate(inputDate) {
     if (!inputDate) return '';
     const {translateLocal} = Localize;
 
-    const input = moment(inputDate, 'YYYY-MM-DD HH:mm:ss');
-    const now = moment();
-    const endOfToday = moment().endOf('day').format('YYYY-MM-DD HH:mm:ss');
+    const input = new Date(inputDate);
+    const now = new Date();
+    const endOfToday = endOfDay(now);
 
     // If the date is equal to the end of today
-    if (input.isSame(endOfToday)) {
+    if (isSameDay(input, endOfToday)) {
```

Comment:
It caused [this bug](https://github.com/Expensify/App/issues/40733) because if we set custom time after 30 minutes, the isSameDay function still returns true